subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -279,6 +199,243 @@ uint8_t ll_sync_create(uint8_t options, uint8_t sid, uint8_t adv_addr_type,
+				       struct ll_sync_set *sync, struct pdu_adv_sync_info *si,
+				       int16_t conn_evt_offset, uint16_t last_pa_event_counter,
+				       uint16_t sync_conn_event_count, uint8_t sender_sca)
+{
+	struct node_rx_past_received *se_past;
+	uint32_t ticks_slot_overhead;
+	uint32_t ticks_slot_offset;
+	uint32_t conn_interval_us;
+	uint32_t sync_offset_us;
+	uint32_t ready_delay_us;
+	struct node_rx_pdu *rx;
+	uint8_t *data_chan_map;
+	struct lll_sync *lll;
+	uint32_t interval_us;
+	uint32_t slot_us;
+	uint32_t ticks_anchor;
+	uint8_t chm_last;
+	uint32_t ret;
+	uint16_t interval;
+	uint16_t sync_handle;
+	uint8_t sca;
+
+	lll = &sync->lll;
+
+	/* Copy channel map from sca_chm field in sync_info structure, and
+	 * clear the SCA bits.
+	 */
+	chm_last = lll->chm_first;
+	lll->chm_last = chm_last;
+	data_chan_map = lll->chm[chm_last].data_chan_map;
+	(void)memcpy(data_chan_map, si->sca_chm,
+		     sizeof(lll->chm[chm_last].data_chan_map));
+	data_chan_map[PDU_SYNC_INFO_SCA_CHM_SCA_BYTE_OFFSET] &=
+		~PDU_SYNC_INFO_SCA_CHM_SCA_BIT_MASK;
+	lll->chm[chm_last].data_chan_count =
+		util_ones_count_get(data_chan_map,
+				    sizeof(lll->chm[chm_last].data_chan_map));
+	if (lll->chm[chm_last].data_chan_count < CHM_USED_COUNT_MIN) {
+		/* Ignore sync setup, invalid available channel count */
+		return;
+	}
+
+	memcpy(lll->access_addr, si->aa, sizeof(lll->access_addr));
+	lll->data_chan_id = lll_chan_id(lll->access_addr);
+	memcpy(lll->crc_init, si->crc_init, sizeof(lll->crc_init));
+	lll->event_counter = sys_le16_to_cpu(si->evt_cntr);
+
+	interval = sys_le16_to_cpu(si->interval);
+	interval_us = interval * PERIODIC_INT_UNIT_US;
+
+	/* Convert fromm 10ms units to interval units */
+	if (sync->timeout != 0  && interval_us != 0) {
+		sync->timeout_reload = RADIO_SYNC_EVENTS((sync->timeout * 10U *
+						  USEC_PER_MSEC), interval_us);
+	}
+
+	/* Adjust Skip value so that there is minimum of 6 events that can be
+	 * listened to before Sync_Timeout occurs.
+	 * The adjustment of the skip value is controller implementation
+	 * specific and not specified by the Bluetooth Core Specification v5.3.
+	 * The Controller `may` use the Skip value, and the implementation here
+	 * covers a case where Skip value could lead to less events being
+	 * listened to until Sync_Timeout. Listening to more consecutive events
+	 * before Sync_Timeout increases probability of retaining the Periodic
+	 * Synchronization.
+	 */
+	if (sync->timeout_reload > CONN_ESTAB_COUNTDOWN) {
+		uint16_t skip_max = sync->timeout_reload - CONN_ESTAB_COUNTDOWN;
+
+		if (sync->skip > skip_max) {
+			sync->skip = skip_max;
+		}
+	}
+
+	sync->sync_expire = CONN_ESTAB_COUNTDOWN;
+
+	/* Extract the SCA value from the sca_chm field of the sync_info
+	 * structure.
+	 */
+	sca = (si->sca_chm[PDU_SYNC_INFO_SCA_CHM_SCA_BYTE_OFFSET] &
+	       PDU_SYNC_INFO_SCA_CHM_SCA_BIT_MASK) >>
+	      PDU_SYNC_INFO_SCA_CHM_SCA_BIT_POS;
+
+	lll->sca = sca;
```

Comment:
dont you need conditional compiles here?

Reply:
The whole function is already flaged with CONFIG_BT_CTLR_SYNC_TRANSFER_RECEIVER.